modules/models/src/main/java/org/eclipse/xpanse/modules/models/service/utils/ServiceVariablesJsonSchemaGenerator.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.modules.models.service.utils;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.networknt.schema.JsonSchema;
+import com.networknt.schema.JsonSchemaFactory;
+import com.networknt.schema.SpecVersion;
+import com.networknt.schema.ValidationMessage;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.xpanse.modules.models.common.exceptions.VariableInvalidException;
+import org.eclipse.xpanse.modules.models.servicetemplate.DeployVariable;
+import org.eclipse.xpanse.modules.models.servicetemplate.utils.JsonObjectSchema;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+@Slf4j
+@Component
+public class ServiceVariablesJsonSchemaGenerator {
+
+    private static final String VARIABLE_TYPE_KEY = "type";
+    private static final String VARIABLE_DESCRIPTION_KEY = "description";
+    private static final String VARIABLE_EXAMPLE_KEY = "example";
+    private final ObjectMapper jsonMapper = new ObjectMapper();
+
+    /**
+     * Check validation of deploy property map by list of deployVariables in registered service.
+     *
+     * @param deployVariables list of deployVariables in registered service
+     * @param deployProperty  deploy property map
+     * @return isValid
+     */
+    public boolean isVariableValid(List<DeployVariable> deployVariables,
```

Comment:
here we never return false. In case of failures, we return exceptions. So better to change return type to void and method name to validateDeployVariables